app/models/course/condition.rb

```diff
@@ -5,4 +5,9 @@ class Course::Condition < ActiveRecord::Base
   belongs_to :conditional, polymorphic: true
 
   delegate :satisfied_by?, to: :actable
+
+  ALL_CONDITIONS = [
+    Course::Condition::Achievement.class_name.downcase,
```

Comment:
```
>> Course::Condition::Achievement.method(:class_name).source_location
["D:/Development/Projects/Coursemology/vendor/bundle/ruby/2.2.0/gems/yard-0.8.7.6/lib/yard/core_ext/module.rb", 7]
```
It's not a Ruby method. It's a Yardoc method.

Keep the conditions as its full class name, transform them later using demodulize, then transform them to route names using underscore.

What happened to my suggestion to use url_for?